test/e2e/prometheus_test.go

```diff
@@ -113,13 +113,38 @@ func runCuratedPackagesPrometheusInstall(test *framework.ClusterE2ETest) {
 		"--set server.persistentVolume.storageClass=local-path")
 	test.VerifyPrometheusPackageInstalled(packageFullName, packageTargetNamespace)
 	test.VerifyPrometheusNodeExporterStates(packageFullName, packageTargetNamespace)
-	test.VerifyPrometheusPrometheusServerStates(packageFullName, packageTargetNamespace)
+	test.VerifyPrometheusPrometheusServerStates(packageFullName, packageTargetNamespace, "deployment")
+}
+
+func runCuratedPackagesPrometheusUpdate(test *framework.ClusterE2ETest) {
+	packageFullName := packagePrefix + "-" + packageName
+
+	test.InstallLocalStorageProvisioner()
+	test.CreateNamespace(packageTargetNamespace)
+	test.SetPackageBundleActive()
+	test.InstallCuratedPackage(packageName, packageFullName,
+		kubeconfig.FromClusterName(test.ClusterName), packageTargetNamespace,
+		"--set server.persistentVolume.storageClass=local-path")
+
+	test.ApplyPrometheusPackageServerStatefulSetFile(packageFullName, packageTargetNamespace)
+	test.VerifyPrometheusPackageInstalled(packageFullName, packageTargetNamespace)
+	test.VerifyPrometheusPrometheusServerStates(packageFullName, packageTargetNamespace, "statefulset")
+	test.VerifyPrometheusNodeExporterStates(packageFullName, packageTargetNamespace)
+
+	test.ApplyPrometheusPackageServerDeploymentFile(packageFullName, packageTargetNamespace)
```

Comment:
Is there any conflicts with both of these installed at the same time? Would it be better to uninstall the other package before moving on to this test?

Reply:
This test was designed to test the switch between statefulset and deployment for prometheus-server (along with other config changes). So, no uninstalling the package first will diminish the purpose of this test.